libsolidity/ast/Types.h

```diff
@@ -357,6 +357,8 @@ class Type
 	/// @returns the canonical name of this type for use in library function signatures.
 	virtual std::string canonicalName() const { return toString(true); }
 	virtual std::string humanReadableName() const { return toString(); }
+	virtual std::string eip712TypeName() const { return encodingType()->toString(true); }
```

Comment:
I wonder if we should leave as a revert?

Or maybe something like
```cpp
solAssert(isEIP712AllowedStructMemberType(), "Invalid type ...");
return encodingType()->toString(true);
```

The reason is that sometimes you get into subtle bugs when new types are introduced due to a default implementation.

Reply:
Agree, this should not be allowed by default for new types